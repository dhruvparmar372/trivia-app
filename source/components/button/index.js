import React, { PureComponent } from "react";
import { View, Platform, TouchableOpacity, StyleSheet } from "react-native";
import Text from "source/components/text";
import { LIGHT_BLUE } from "source/constants/colors";

class Button extends PureComponent {
  render() {
    const {
      rounded,
      backgroundColor = LIGHT_BLUE,
      text,
      textColor,
      style = {},
      ...rest
    } = this.props;

    return (
      <TouchableOpacity
        {...rest}
        style={[
          styles.container,
          { backgroundColor },
          rounded ? { borderRadius: 2 } : {},
          style
        ]}
      >
        <Text style={[styles.text, { color: textColor }]}>{text}</Text>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    padding: 12
  },
  text: {
    textAlign: "center"
  }
});

export default Button;
