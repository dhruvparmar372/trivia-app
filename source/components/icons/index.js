import React from "react";
import { Platform } from "react-native";
import MaterialIcon from "react-native-vector-icons/MaterialIcons";
import IoniIcon from "react-native-vector-icons/Ionicons";
import EntypoIcon from "react-native-vector-icons/Entypo";

export const IconBack = props =>
  Platform.OS === "ios" ? (
    <IoniIcon name={"ios-arrow-back"} {...props} />
  ) : (
    <IoniIcon name={"md-arrow-back"} {...props} />
  );

export const IconCorrect = props =>
  Platform.OS === "ios" ? (
    <IoniIcon name={"ios-checkmark-circle"} {...props} />
  ) : (
    <MaterialIcon name={"check-circle"} {...props} />
  );

export const IconWrong = props =>
  Platform.OS === "ios" ? (
    <IoniIcon name={"ios-close-circle"} {...props} />
  ) : (
    <MaterialIcon name={"cancel"} {...props} />
  );

export const IconHappy = props => (
  <EntypoIcon name={"emoji-happy"} {...props} />
);
export const IconNeutral = props => (
  <EntypoIcon name={"emoji-neutral"} {...props} />
);
export const IconSad = props => <EntypoIcon name={"emoji-sad"} {...props} />;
