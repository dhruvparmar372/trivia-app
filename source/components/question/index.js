import React, { PureComponent } from "react";
import { View, StyleSheet, TouchableOpacity } from "react-native";
import _shuffle from "lodash.shuffle";
import { AllHtmlEntities as Entities } from "html-entities";
import I18n from "react-native-i18n";
import Text from "source/components/text";
import { GRAY, LIGHT_BLUE, DARK_BLUE } from "source/constants/colors";

const entities = new Entities();
class Question extends PureComponent {
  recordAnswer = answer => {
    this.props.onAnswer({ value: answer });
  };

  render() {
    const {
      question: {
        category,
        question: questionText,
        incorrect_answers,
        correct_answer
      }
    } = this.props;
    const answers = _shuffle([...incorrect_answers, correct_answer]);

    return (
      <View>
        <Text style={styles.category}>{category}</Text>
        <View style={styles.question}>
          <Text style={styles.questionText}>
            {entities.decode(questionText)}
          </Text>
        </View>
        {answers.map(answer => (
          <TouchableOpacity
            key={answer}
            onPress={() => this.recordAnswer(answer)}
            style={styles.answer}
          >
            <Text style={styles.answerText}>{answer}</Text>
          </TouchableOpacity>
        ))}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  category: {
    fontSize: 18,
    color: GRAY,
    marginBottom: 15
  },
  question: {
    borderWidth: 1,
    borderColor: GRAY,
    padding: 10,
    borderRadius: 2,
    marginBottom: 24
  },
  questionText: {
    fontSize: 29,
    color: GRAY
  },

  answer: {
    backgroundColor: LIGHT_BLUE,
    padding: 12,
    borderRadius: 2,
    marginTop: 10
  },
  answerText: {
    color: DARK_BLUE,
    fontSize: 18,
    textAlign: "center"
  }
});

export default Question;
