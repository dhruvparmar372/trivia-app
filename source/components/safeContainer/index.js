import React from "react";
import { SafeAreaView } from "react-navigation";
import { StatusBar, StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1
  }
});

export default function(Component, { backgroundColor } = {}) {
  return function(props) {
    return (
      <SafeAreaView
        style={[styles.container, backgroundColor ? { backgroundColor } : {}]}
      >
        <Component {...props} />
      </SafeAreaView>
    );
  };
}
