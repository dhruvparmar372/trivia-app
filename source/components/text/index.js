import React from "react";
import { Text as RNText, StyleSheet } from "react-native";

const Text = props => (
  <RNText {...props} style={[styles.text].concat(props.style)}>
    {props.children}
  </RNText>
);

Text.propTypes = RNText.propTypes;

const styles = StyleSheet.create({
  text: {
    fontFamily: "System",
    fontSize: 16
  }
});

export default Text;
