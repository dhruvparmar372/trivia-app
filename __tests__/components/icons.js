import React from "react";
import renderer from "react-test-renderer";
import { IconBack, IconCorrect, IconWrong } from "source/components/icons";

describe("icons android", () => {
  beforeAll(() => {
    jest.mock("Platform", () => {
      const Platform = require.requireActual("Platform");
      Platform.OS = "android";
      return Platform;
    });
  });

  test("back icon", () => {
    expect(renderer.create(<IconBack size={24} />)).toMatchSnapshot();
  });

  test("correct icon", () => {
    expect(renderer.create(<IconCorrect size={24} />)).toMatchSnapshot();
  });

  test("wrong icon", () => {
    expect(renderer.create(<IconWrong size={24} />)).toMatchSnapshot();
  });
});

describe("icons ios", () => {
  beforeAll(() => {
    jest.mock("Platform", () => {
      const Platform = require.requireActual("Platform");
      Platform.OS = "ios";
      return Platform;
    });
  });

  test("back icon", () => {
    expect(renderer.create(<IconBack size={24} />)).toMatchSnapshot();
  });

  test("correct icon", () => {
    expect(renderer.create(<IconCorrect size={24} />)).toMatchSnapshot();
  });

  test("wrong icon", () => {
    expect(renderer.create(<IconWrong size={24} />)).toMatchSnapshot();
  });
});
