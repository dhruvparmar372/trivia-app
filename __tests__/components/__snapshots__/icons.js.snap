// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icons android back icon 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`icons android correct icon 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`icons android wrong icon 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`icons ios back icon 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`icons ios correct icon 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`icons ios wrong icon 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;
